### PR TITLE
handle aiohttp errors

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py [hunk 1]
```diff
@@ -324,6 +324,12 @@ async def __anext__(self):
             _LOGGER.warning("Incomplete download: %s", err)
             internal_response.close()
             raise IncompleteReadError(err, error=err)
+        except aiohttp.client_exceptions.ClientResponseError as err:
+            raise ServiceResponseError(err, error=err) from err
+        except aiohttp.client_exceptions.ClientError as err:
+            raise ServiceRequestError(err, error=err) from err
+        except asyncio.TimeoutError as err:
```
Comment (kashifkhan) [inline sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py:331]: small change to swap the position of `asyncio.TimeoutError` with `ClientError` as the latter is the base class for all aiohttp errors.

**Comment by kashifkhan** [comment]
will an update be needed for the changelog ? This can affect folks who had workaround for the previous errors

**Comment by xiangyan99** [comment]
> will an update be needed for the changelog ? This can affect folks who had workaround for the previous errors

Yes. If the change looks good to you, I will push another change to update changelog. Hold it on to avoid merging conflicts. :)

**Comment by kashifkhan** [comment]
> > will an update be needed for the changelog ? This can affect folks who had workaround for the previous errors
> 
> Yes. If the change looks good to you, I will push another change to update changelog. Hold it on to avoid merging conflicts. :)

yes the change looks good to me :)

**Comment by xiangyan99** [comment]
> > > will an update be needed for the changelog ? This can affect folks who had workaround for the previous errors
> > 
> > 
> > Yes. If the change looks good to you, I will push another change to update changelog. Hold it on to avoid merging conflicts. :)
> 
> yes the change looks good to me :)

Changelog updated. :)

**Review by kashifkhan — APPROVED** [review]

**Pull request merged**